checks/http_test.go

```diff
@@ -5,17 +5,48 @@ import (
 	"net/http"
 	"net/http/httptest"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
 const (
-	longRequest     = "LongRequest"
-	expectedContent = "I'm healthy"
+	longRequest       = "LongRequest"
+	expectedContent   = "I'm healthy"
+	testCookieKey     = "test-cookie"
+	expectedCookieVal = "test-cookie-val"
+	testHeaderKey     = "test-header"
+	expectedHeaderVal = "test-value"
 )
 
+type receivedRequest struct {
+	sync.RWMutex
+	details map[string]string
+}
+
+func (req *receivedRequest) addDetail(key string, val string) {
+	if val == "" {
+		return
+	}
+	req.Lock()
+	defer req.Unlock()
+	req.details[key] = val
+}
+
+func (req *receivedRequest) getDetail(key string) string {
+	req.RLock()
+	defer req.RUnlock()
+	return req.details[key]
+}
+
+func (req *receivedRequest) clear() {
+	req.Lock()
+	defer req.Unlock()
+	req.details = make(map[string]string, 2)
```

Comment:
why `2`?

Reply:
It's just a test... I make up to 2 changes in the test that uses that...